### PR TITLE
feat: Handle guest link feature flag

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

This PR aims to handle guest link feature flag by disabling/enabling it if it's changed from admin console.

A dialog will be displayed if a config change is detected 

### Dependencies (Optional)

Needs to be merged after [feat: add feature flag for guest room link (AR-2452)](https://github.com/wireapp/kalium/pull/1524)

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

When you disable/enable this feature from admin console you should see a popup informing you about this change.
And Create guest link button will be disabled

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5811452)(5811452)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5811452)(5811452)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/6/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/8/) **failed**.

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/10/) **failed**.

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-1488/12/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5861729)(5861729)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5861729)(5861729)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/Buttons.kt [hunk 1]
```diff
@@ -46,7 +46,8 @@ fun CreateLinkButton(
         fillMaxWidth = true,
         onClick = onCreateLink,
         loading = isLoading,
-        state = if (isLoading || !isGuestAccessAllowed) WireButtonState.Disabled else WireButtonState.Default,
+        state = if (shouldDisableGenerateGuestLinkButton) WireButtonState.Disabled
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/Buttons.kt:49]: Shouldn't we keep accounting for `isLoading` as well?

File: app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt [hunk 2]
```diff
@@ -23,5 +23,7 @@ package com.wire.android.ui.home
 data class FeatureFlagState(
     val showFileSharingDialog: Boolean = false,
     val isFileSharingEnabledState: Boolean = true,
-    val showFileSharingRestrictedDialog: Boolean = false
+    val showFileSharingRestrictedDialog: Boolean = false,
+    val shouldShowGuestRoomLinkDialog: Boolean = false,
+    val isGuestRoomLinkEnabled: Boolean = true
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt:28]: Should this be enabled by default? Shouldn't it start as false? 🤔

File: default.json [hunk 3]
```diff
@@ -83,5 +83,6 @@
     "force_constant_bitrate_calls": false,
     "mls_support_enabled": true,
     "encrypt_proteus_storage": false,
-    "update_app_url": "https://wire.com/en/download/"
+    "update_app_url": "https://wire.com/en/download/",
+    "guest_room_link": true
```
Comment (alexandreferris) [inline default.json:87]: Same question as above:
`Should this be enabled by default? Shouldn't it start as false? 🤔 `

**Review by alexandreferris — COMMENTED** [review]
Great work @ohassine ! Specially on moving code to specific classes/files!

I just have some small comments/questions mostly.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/Buttons.kt (hunk 1 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/Buttons.kt:49]: yes we are considering it in the viewModel, `isGeneratingGuestRoomLink`

```
fun shouldDisableGenerateGuestLinkButton() = !editGuestAccessState.isGuestRoomLinkFeatureEnabled ||
            !editGuestAccessState.isGuestAccessAllowed ||
            editGuestAccessState.isGeneratingGuestRoomLink
```

File: app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt (hunk 2 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt:28]: No guest link are enabled by default unless a team wants to disable it.

File: default.json (hunk 3 above)
Comment (ohassine) [reply default.json:87]: same as above 😄

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/Buttons.kt (hunk 1 above)
Comment (alexandreferris) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/Buttons.kt:49]: Aha! Read it through and it went past me, sorry!

**Review by alexandreferris — APPROVED** [review]
🔥

**Pull request merged**